internal/dlv/delve.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dlv
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-delve/delve/cmd/dlv/cmds"
+)
+
+// MainWithArgs is a type representing a function that takes command-line arguments as input and returns an exit code.
+type MainWithArgs func(args []string) int
+
+const (
+	// envNoDebug is an environment variable used to redirect the control flow to a new delve instance
+	// or to the program. It is set as environment variable the first time the flow pass in the function
+	envNoDebug = "JUJU_NO_DLV_DEBUG"
+)
+
+// NewDlvRunner wraps a MainWithArgs function to enable debugging.
+//
+// It works in two phases:
+// At first call, it launches the delve command `exec` with the current binary. However, before the
+// call it set a environment variable DELVE_ANYTHING_NO_DEBUG.
```

Comment:
"it sets".

Reply:
Done https://github.com/juju/juju/pull/18114/commits/8735fc258e9495c527885ea569ef7368901991d6